[gameplay]/voice_local/client.lua

```diff
@@ -1,155 +1,144 @@
-        setSoundVolume(localPlayer, 0)
-        triggerServerEvent("voice:removePlayerBroadcast", resourceRoot, localPlayer, localPlayer)
-    end
-end
-addEventHandler("onClientPlayerWasted", localPlayer, onWasted)
+addEventHandler("onClientElementStreamIn", root, function()
+    if source == localPlayer then return end
+    if not (isElement(source) and getElementType(source) == "player") then return end
 
-function resourceStop()
-    triggerServerEvent("voice:removePlayerBroadcasts", resourceRoot, localPlayer, streamedPlayers)
-    streamedPlayers = {}
-end
-addEventHandler("onClientResourceStop", resourceRoot, resourceStop, false)
+    if isPedDead(source) then return end
 
-function voiceStart(source)
-    if (isElement(source) and getElementType(source) == "player") then
-        if streamedPlayers[source] ~= nil then
-            streamedPlayers[source] = true
-        end
+    if streamedPlayers[source] == nil then
+        setSoundVolume(source, 0)
+        streamedPlayers[source] = false
+        triggerServerEvent("voice_local:addToPlayerBroadcast", resourceRoot, source)
     end
-end
-addEvent("voice_cl:onClientPlayerVoiceStart", true)
-addEventHandler("voice_cl:onClientPlayerVoiceStart", resourceRoot, voiceStart)
+end)
+addEventHandler("onClientElementStreamOut", root, function()
+    if source == localPlayer then return end
+    if not (isElement(source) and getElementType(source) == "player") then return end
 
-function voiceStop(source)
-    if (isElement(source) and getElementType(source) == "player") then
-        if streamedPlayers[source] ~= nil then
-            streamedPlayers[source] = false
-        end
+    if streamedPlayers[source] ~= nil then
+        setSoundVolume(source, 0)
+        streamedPlayers[source] = nil
+        triggerServerEvent("voice_local:removeFromPlayerBroadcast", resourceRoot, source)
     end
-end
-addEvent("voice_cl:onClientPlayerVoiceStop", true)
-addEventHandler("voice_cl:onClientPlayerVoiceStop", resourceRoot, voiceStop)
+end)
 
-function table.empty(a)
-    if type(a) ~= "table" then
-        return false
+-- Update player talking status (for displaying)
+addEventHandler("voice_local:onClientPlayerVoiceStart", resourceRoot, function(player)
+    if not (isElement(player) and getElementType(player) == "player") then return end
+
+    if player == localPlayer then
+        localPlayerTalking = true
+    elseif streamedPlayers[player] ~= nil then
+        streamedPlayers[player] = true
+    end
+end)
+addEventHandler("voice_local:onClientPlayerVoiceStop", resourceRoot, function(player)
+    if not (isElement(player) and getElementType(player) == "player") then return end
+
+    if player == localPlayer then
+        localPlayerTalking = false
+    elseif streamedPlayers[player] ~= nil then
+        streamedPlayers[player] = false
     end
+end)
 
-    return next(a) == nil
-end
\ No newline at end of file
+addEventHandler("voice_local:updateSettings", resourceRoot, function(settingsFromServer)
```

Comment:
```suggestion
addEventHandler("voice_local:updateSettings", localPlayer, function(settingsFromServer)
```

In this case event is sent only to localPlayer